src/core/algorithms/algorithm_types.h

```diff
@@ -10,7 +10,7 @@ using AlgorithmTypes =
         std::tuple<Depminer, DFD, FastFDs, FDep, Fd_mine, Pyro, Tane, FUN, hyfd::HyFD, Aid, Apriori,
                    metric::MetricVerifier, DataStats, fd_verifier::FDVerifier, HyUCC, PyroUCC,
                    cfd::FDFirstAlgorithm, ACAlgorithm, UCCVerifier, Faida, Spider, GfdValidation,
-                   EGfdValidation, NaiveGfdValidation>;
+                   fastod::Fastod, EGfdValidation, NaiveGfdValidation>;
```

Comment:
`fastod::Fastod` type should be after `Spider` type in this tuple (as in enum `AlgorithmType`). Because in the Algo Factory we have method `CreateAlgorithm`, that receives `AlgorithmType` as parameter and takes the enum value from it (in your case AlgorithmType::fastod is 21) and then we take the corresponding type from tuple. And in your case we will get incorrect type.
At the moment nothing is broken since we are not explicitly calling this method, but we should fix this

Reply:
Did it.